Development/nmos/api_downgrade.cpp

```diff
@@ -68,69 +1030,219 @@ namespace nmos
         return downgrade(resource.version, resource.downgrade_version, resource.type, resource.data, version, downgrade_version);
     }
 
-    static const std::map<nmos::type, std::map<nmos::api_version, std::vector<utility::string_t>>>& resources_versions()
+    struct property
+    {
+        utility::string_t name;
+        web::json::value default_value;
+        bool required;  // is required property
+
+        property(utility::string_t name, web::json::value default_value = web::json::value::null(), bool required = true)
+            : name(std::move(name))
+            , default_value(std::move(default_value))
+            , required(std::move(required))
+        {}
+    };
+
+    static const std::map<nmos::type, std::map<nmos::api_version, std::vector<property>>>& resources_versions()
     {
-        static const std::map<nmos::type, std::map<nmos::api_version, std::vector<utility::string_t>>> resources_versions
+        using web::json::value;
+        using web::json::value_of;
+
+        static const std::map<nmos::type, std::map<nmos::api_version, std::vector<property>>> resources_versions
         {
             {
                 nmos::types::node,
                 {
-                    { nmos::is04_versions::v1_0, { U("id"), U("version"), U("label"), U("href"), U("hostname"), U("caps"), U("services") } },
-                    { nmos::is04_versions::v1_1, { U("description"), U("tags"), U("api"), U("clocks") } },
-                    { nmos::is04_versions::v1_2, { U("interfaces") } }
+                    { nmos::is04_versions::v1_0, {
+                        { U("id") },
+                        { U("version") },
+                        { U("label") },
+                        { U("href") },
+                        { U("hostname") },
+                        { U("caps") },
+                        { U("services"), value::array() }
```

Comment:
This doesn't seem like right strategy to me. I don't think we should remove `authorization` when considering v1.3 node resource and exposing it via Node API v1.0 or registering via Registration API v1.0. If someone has good reason that we should do that, I don't think we should ever remove all services and replace with empty array... 😕

Reply:
as long as the JSON validation is successful, the `authorization` parameter will not be removed.